Bottom: c254f8b910282028c7a754aa6e5938952d7c4a9f
Top:    69bc80db6e3360c69b2be2547bfb9aa3137c0156
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-04 20:01:39 +0800

Refresh of tuntap-switch-to-use

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 38574cd..2290639 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1576,8 +1576,7 @@ static void tun_rx_batched(struct tun_struct *tun, struct tun_file *tfile,
 
 	spin_lock(&queue->lock);
 	if (!more || skb_queue_len(queue) == rx_batched) {
-		list = tfile->rx_list;
-		INIT_LIST_HEAD(&tfile->rx_list);
+		list_splice_tail_init(&tfile->rx_list, &list);
 		rcv = true;
 	} else {
 		list_add_tail(&skb->list, &tfile->rx_list);
